temporalio/workflow.py

```diff
@@ -5074,3 +5152,71 @@ def _to_proto(self) -> temporalio.bridge.proto.common.VersioningIntent.ValueType
         elif self == VersioningIntent.DEFAULT:
             return temporalio.bridge.proto.common.VersioningIntent.DEFAULT
         return temporalio.bridge.proto.common.VersioningIntent.UNSPECIFIED
+
+
+# Nexus
+
+ServiceT = TypeVar("ServiceT")
+
+
+class NexusClient(Generic[ServiceT]):
+    def __init__(
+        self,
+        service: Union[Type[ServiceT], str],
+        *,
+        endpoint: str,
+    ) -> None:
+        # If service is not a str, then it must be a service interface or implementation
+        # class.
+        if isinstance(service, str):
+            self._service_name = service
+        # TODO(preview): make double-underscore attrs private to nexusrpc and expose getters/setters
+        elif service_defn := getattr(service, "__nexus_service__", None):
+            self._service_name = service_defn.name
+        else:
+            raise ValueError(
+                f"`service` may be a name (str), or a class decorated with either "
+                f"@nexusrpc.handler.service_handler or @nexusrpc.service. "
+                f"Invalid service type: {type(service)}"
+            )
+        self._endpoint = endpoint
+
+    # TODO(nexus-prerelease): overloads: no-input, ret type
+    # TODO(nexus-prerelease): should it be an error to use a reference to a method on a class other than that supplied?
+    async def start_operation(
+        self,
+        operation: Union[nexusrpc.Operation[InputT, OutputT], str, Callable[..., Any]],
+        input: InputT,
+        *,
+        output_type: Optional[Type[OutputT]] = None,
+        schedule_to_close_timeout: Optional[timedelta] = None,
+        headers: Optional[Mapping[str, str]] = None,
+    ) -> NexusOperationHandle[OutputT]:
+        return await temporalio.workflow.start_nexus_operation(
+            endpoint=self._endpoint,
+            service=self._service_name,
+            operation=operation,
+            input=input,
+            output_type=output_type,
+            schedule_to_close_timeout=schedule_to_close_timeout,
+            headers=headers,
+        )
+
+    # TODO(nexus-prerelease): overloads: no-input, ret type
+    async def execute_operation(
+        self,
+        operation: Union[nexusrpc.Operation[InputT, OutputT], str, Callable[..., Any]],
+        input: InputT,
+        *,
+        output_type: Optional[Type[OutputT]] = None,
+        schedule_to_close_timeout: Optional[timedelta] = None,
+        headers: Optional[Mapping[str, str]] = None,
+    ) -> OutputT:
+        handle: NexusOperationHandle[OutputT] = await self.start_operation(
```

Comment:
```suggestion
        handle = await self.start_operation(
```

Just curious if this is required (don't have to change anything, just confirming type checkers are smart enough when a user does this themselves)

Reply:
Good catch. This must have been necessary at some point otherwise I wouldn't have put it there, but the types work now! I certainly don't want it there. Removed.